pyuvdata/uvbeam/beamfits.py

```diff
@@ -193,8 +215,87 @@ def read_beamfits(
 
+            # shapes
+            if (
+                primary_header.pop("CTYPE" + str(ax_nums["freq"])).lower().strip()
+                == "freq"
+            ):
+                self.Nfreqs = primary_header.pop("NAXIS" + str(ax_nums["freq"]))
+
+            if n_dimensions > ax_nums["spw"] - 1:
+                if (
+                    primary_header.pop("CTYPE" + str(ax_nums["spw"])).lower().strip()
+                    == "if"
+                ):
+                    self.Nspws = primary_header.pop("NAXIS" + str(ax_nums["spw"]), None)
+                    # subtract 1 to be zero-indexed
+                    self.spw_array = (
+                        uvutils._fits_gethduaxis(primary_hdu, ax_nums["spw"]) - 1
+                    )
+
+            if n_dimensions > ax_nums["basisvec"] - 1:
+                if (
+                    primary_header.pop("CTYPE" + str(ax_nums["basisvec"]))
+                    .lower()
+                    .strip()
+                    == "vecind"
+                ):
+                    self.Naxes_vec = primary_header.pop(
+                        "NAXIS" + str(ax_nums["basisvec"]), None
+                    )
+
+            if (
+                self.Nspws is None or self.Naxes_vec is None
+            ) and self.beam_type == "power":
+                if self.Nspws is None:
+                    self.Nspws = 1
+                    self.spw_array = np.array([0])
+                if self.Naxes_vec is None:
+                    self.Naxes_vec = 1
+
+                # add extra empty dimensions to data_array as appropriate
+                while len(data.shape) < n_efield_dims - 1:
+                    data = np.expand_dims(data, axis=0)
+
+            if self.Nspws is None:
+                self.Nspws = 1
+
+            self.freq_array = uvutils._fits_gethduaxis(primary_hdu, ax_nums["freq"])
+            self.freq_array.shape = (self.Nspws,) + self.freq_array.shape
+            # default frequency axis is Hz, but check for corresonding CUNIT
+            freq_units = primary_header.pop("CUNIT" + str(ax_nums["freq"]), "Hz")
+            if freq_units != "Hz":
+                freq_factor = {"kHz": 1e3, "MHz": 1e6, "GHz": 1e9}
+                if freq_units in freq_factor.keys():
+                    self.freq_array = self.freq_array * freq_factor[freq_units]
+                else:
+                    raise ValueError("Frequency units not recognized.")
+
+            if freq_range is not None:
+                fmin = np.where(self.freq_array[0] >= freq_range[0])[0][0]
+                fmax = np.where(self.freq_array[0] <= freq_range[1])[0][-1] + 1
+                freq_mask = slice(fmin, fmax)
+                self.Nfreqs = fmax - fmin
+                self.freq_array = self.freq_array[:, freq_mask]
+
+            if az_range is not None:
+                data = data[..., az_mask]
+            if za_range is not None:
+                data = data[..., za_mask, :]
```

Comment:
I can't quite tell the context here, will this run for healpix files? If so, we need to error if these keywords are set for healpix files.

Reply:
It will *not* run for healpix! I'll add an error -- good catch!